pkg/controllers/user/networkpolicy/netpol.go

```diff
@@ -313,36 +182,41 @@ func (npmgr *netpolMgr) handleHostNetwork(clusterNamespace string) error {
 	return nil
 }
 
-func (npmgr *netpolMgr) getSystemNamespaces(clusterNamespace string) (map[string]bool, error) {
+func (npmgr *netpolMgr) getSystemNamespaces(clusterNamespace string) (map[string]bool, string, error) {
 	systemNamespaces := map[string]bool{}
 	set := labels.Set(map[string]string{systemProjectLabel: "true"})
 	projects, err := npmgr.projLister.List(clusterNamespace, set.AsSelector())
+	systemProjectID := ""
 	if err != nil {
-		return nil, err
+		return nil, systemProjectID, err
 	}
 	if len(projects) == 0 {
-		return systemNamespaces, fmt.Errorf("systemNamespaces: no system project for cluster %s", clusterNamespace)
+		return systemNamespaces, systemProjectID,
+			fmt.Errorf("systemNamespaces: no system project for cluster %s", clusterNamespace)
 	}
 	if len(projects) > 1 {
-		return systemNamespaces, fmt.Errorf("systemNamespaces: more than one system project in cluster %s", clusterNamespace)
+		return systemNamespaces, systemProjectID,
+			fmt.Errorf("systemNamespaces: more than one system project in cluster %s", clusterNamespace)
 	}
-	systemProjectID := projects[0].Name
-	if systemProjectID != "" {
-		set := labels.Set(map[string]string{nslabels.ProjectIDFieldLabel: systemProjectID})
-		namespaces, err := npmgr.nsLister.List("", set.AsSelector())
-		if err != nil {
-			return nil, fmt.Errorf("sytemNamespaces: couldn't list namespaces err=%v", err)
-		}
-		for _, ns := range namespaces {
-			if _, ok := systemNamespaces[ns.Name]; !ok {
-				systemNamespaces[ns.Name] = true
-			}
+	systemProjectID = projects[0].Name
```

Comment:
A quick comment explaining the translation here from name to id would be helpful in future